core/src/main/java/com/aembootcamp/core/models/ArticleModel.java

```diff
@@ -0,0 +1,90 @@
+package com.aembootcamp.core.models;
+
+import lombok.Getter;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.DefaultInjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.SlingObject;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+
+import com.adobe.cq.dam.cfm.ContentFragment;
+import com.day.cq.wcm.api.Page;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+
+@Getter
+@Model(adaptables = { Resource.class, SlingHttpServletRequest.class }, resourceType = {
+        ArticleModel.RESOURCE_TYPE }, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class ArticleModel {
+
+    protected static final String RESOURCE_TYPE = "aem-bootcamp/components/structure/article";
+
+    @ValueMapValue
+    private String title;
+
+    @ValueMapValue
+    private String fileReference;
+
+    @ValueMapValue
+    private String description;
+
+    @ValueMapValue
+    private String authorContentFragment;
+
+    @ValueMapValue
+    private String[] categories;
+
+    @SlingObject
+    private ResourceResolver resourceResolver;
+
+    @SlingObject
+    private Resource resource;
+
+    @Inject
+    private Page currentPage;
+
+    private String authorName;
+
+    private String authorDisplayPicture;
+
+    private String authorBio;
+
+    private String categoriesTagText;
+
+    @PostConstruct
+    private void init() {
+        fetchTagText();
+        fetchTitle();
+        fetchContentFragmentData();
+    }
+
+    private void fetchTagText() {
+        if (currentPage != null) {
+            if (currentPage.getProperties().get("articlecategories") != null) {
```

Comment:
You don't need the current page, please use the field categories. You can do something like this:
```
@ValueMapValue(name="articlecategories")
private String[] categories;
```

The other option would be to use the `@Named` annotation
```
@ValueMapValue
@Named("articlecategories")
private String[] categories;
```

Or you could simply rename the categories field to articleCategories and do something like this:
```
@ValueMapValue
private String[] articleCategories;
```
Please verify if you also need to rename the field in the dialog to be camel case.